homeassistant/setup.py

```diff
@@ -89,7 +85,8 @@ async def _async_process_dependencies(
         return True
 
     _LOGGER.debug("Dependency %s will wait for %s", integration.domain, list(tasks))
-    results = await asyncio.gather(*tasks.values())
+    async with hass.timeout.freeze(integration.domain):
```

Comment:
This is where we hit the should not happen case in tests and it ends up `_FreezeGlobal` because we will have never set the timeout.